File: scripts/colormap.py
import cv2
from cv_bridge import CvBridge
import rclpy
from rclpy.node import Node
from sensor_msgs.msg import Image

import time

class ColorMap(Node):
  def __init__(self):
    super().__init__("colormap_node")

    self._sub = self.create_subscription(Image, "/flir/ax5_camera_node/image_raw", self.color_image, 10)
    self._pub = self.create_publisher(Image, "/flir/ax5_camera_node/image_colormap", 10)
    self._cvb = CvBridge()

  def color_image(self, msg):
    tic = time.perf_counter()
    cvimage = self._cvb.imgmsg_to_cv2(msg)
    new_img = cv2.applyColorMap(cvimage, cv2.COLORMAP_PLASMA)
    msgimg: Image = self._cvb.cv2_to_imgmsg(new_img, 'bgr8')
    # cv2.imshow("debug", new_img)
    # cv2.waitKey(0)
    # print(msgimg.encoding)
    self._pub.publish(msgimg)
    print(time.perf_counter() - tic)

if __name__ == '__main__':
  rclpy.init()
  node = ColorMap()
  rclpy.spin(node)

File: src/colormap.cpp
#include <opencv2/core.hpp>
#include <opencv2/imgproc.hpp>
#include <opencv2/imgcodecs.hpp>

#include <cv_bridge/cv_bridge.h>

#include <rclcpp/rclcpp.hpp>
#include <sensor_msgs/msg/image.hpp>

#include <string>
#include <chrono>

class ColorMapNode : public rclcpp::Node
{
public:
	ColorMapNode() 
	: rclcpp::Node("colormap_node")
	{
		pub_ = this->create_publisher<sensor_msgs::msg::Image>("ax5_camera_node/image_colormap", 1);
		sub_ = this->create_subscription<sensor_msgs::msg::Image>("ax5_camera_node/image_raw", 1, 
																  [this](const sensor_msgs::msg::Image& msg) -> void {
																			std::chrono::time_point tic = std::chrono::steady_clock::now();
																			this->_cvb = cv_bridge::toCvCopy(msg, "bgr8");
																			cv::applyColorMap(this->_cvb->image, this->_cvb->image, cv::COLORMAP_MAGMA);
																			this->_cvb->toImageMsg(to_send_);
																			this->pub_->publish(to_send_);
																			RCLCPP_DEBUG_STREAM(this->get_logger(), (std::chrono::steady_clock::now() - tic).count());
																  }
		);
	}
private:
	rclcpp::Publisher<sensor_msgs::msg::Image>::SharedPtr pub_;
	rclcpp::Subscription<sensor_msgs::msg::Image>::SharedPtr sub_;
	cv_bridge::CvImagePtr _cvb;
	sensor_msgs::msg::Image to_send_;
};

int main(int argc, char** argv)
{
	rclcpp::init(argc, argv);
	auto node = std::make_shared<ColorMapNode>();
	rclcpp::spin(node);
	
	return 0;
}
